src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,246 @@
+            // the initial restore, so we restore the packages ourselves.
+            await DotnetMSBuild("Restore");
+
+            var result = await DotnetMSBuild("GenerateAspNetCoreStaticAssetsManifest");
+
+            Assert.BuildPassed(result);
+
+            // GenerateAspNetCoreStaticAssetsManifest should generate the manifest and the cache.
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.xml");
+            Assert.FileExists(result, IntermediateOutputPath, "Microsoft.AspNetCore.StaticAssets.cache");
+
+            var directoryPath = Path.Combine(result.Project.DirectoryPath, IntermediateOutputPath);
+            var thumbPrints = new Dictionary<string, FileThumbPrint>();
+            var thumbPrintFiles = new[]
+            {
+                Path.Combine(directoryPath, "Microsoft.AspNetCore.StaticAssets.xml"),
+                Path.Combine(directoryPath, "Microsoft.AspNetCore.StaticAssets.cache"),
+            };
+
+            foreach (var file in thumbPrintFiles)
+            {
+                var thumbprint = GetThumbPrint(file);
+                thumbPrints[file] = thumbprint;
+            }
+
+            // Act
+            var incremental = await DotnetMSBuild("GenerateAspNetCoreStaticAssetsManifest");
+
+            // Assert
+            Assert.BuildPassed(incremental);
+
+            foreach (var file in thumbPrintFiles)
+            {
+                var thumbprint = GetThumbPrint(file);
+                Assert.Equal(thumbPrints[file], thumbprint);
+            }
+        }
+    }
+
+    public class PackageTestProjectsFixture
+    {
+        private bool _packed;
+
+        internal void Pack(ITestOutputHelper output)
+        {
+            if (_packed)
+            {
+                return;
+            }
+
+            var projectsToPack = typeof(PackageTestProjectsFixture).Assembly.GetCustomAttributes<AssemblyMetadataAttribute>()
+                .Where(a => a.Key == "Testing.ProjectToPack")
+                .Select(a => a.Value)
+                .ToArray();
+
+            foreach (var project in projectsToPack)
+            {
+                output.WriteLine(project);
+            }
+
+            foreach (var project in projectsToPack)
+            {
+                var psi = new ProcessStartInfo
+                {
+                    FileName = DotNetMuxer.MuxerPathOrDefault(),
+#if DEBUG
+                    Arguments = "pack -c Debug",
+#else
+                    Arguments = "pack -c Release",
+#endif
+                    WorkingDirectory = project,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                };
+                var process = Process.Start(psi);
+
+                // Wait for 10 seconds or fail. If we take longer
+                // it likely means we are adding unexpected dependencies.
+                Assert.True(process.WaitForExit(180 * 1000));
```

Comment:
Ideally this would be async. We don't want to block in tests because it will cause thread starvation.